docs/concepts/workloads/pods/disruptions.md

```diff
@@ -0,0 +1,262 @@
+| pod-x  *terminating* |                     |                    |
+
+The deployment notices that one of the pods is terminating, so it creates a replacement,
+called `pod-d`.  Since `node-1` is cordoned, it lands on another node.  Something has
+also created `pod-y` as a replacement for `pod-x`.
+
+(Note: for a StatefulSet, `pod-a`, which would be called something like `pod-1`, would need
+to terminate completely before its replacement, which is also called `pod-1 but has a
+different UID, could be created.  Otherwise, the example applies to a StatefulSet as well.)
+
+Now the cluster is in this state:
+
+|   node-1 *draining*  |       node-2        |       node-3       |
+|:--------------------:|:-------------------:|:------------------:|
+| pod-a  *terminating* | pod-b *available*   | pod-c *available*  |
+| pod-x  *terminating* | pod-d *starting*    | pod-y              |
+
+At some point, the pods terminate, and the cluster look like this:
+
+|    node-1 *drained*  |       node-2        |       node-3       |
+|:--------------------:|:-------------------:|:------------------:|
+|                      | pod-b *available*   | pod-c *available*  |
+|                      | pod-d *starting*    | pod-y              |
+
+At this point, if an impatient cluster administrator tries to drain `node-2` or
+`node-3`, they will not succeed, because there are only 2 available pods for the deployment, 
+and its PDB requires at least 2.  After some time passes, `pod-d` becomes available.
+
+The cluster state now looks like this:
+
+|    node-1 *drained*  |       node-2        |       node-3       |
+|:--------------------:|:-------------------:|:------------------:|
+|                      | pod-b *available*   | pod-c *available*  |
+|                      | pod-d *available*   | pod-y              |
+
+Now, the cluster admin tries to drain `node-2`.
+The drain command will try to evict the two pods in some order, say 
+`pod-b` first and then `pod-d`.  It will succeed at evicting `pod-b`.
+But, when it tries to evict `pod-d`, it will be refused because that would leave only
+one pod available for the deployment.
+
+The deployment creates a replacement for `pod-b` called `pod-e`.
+However, not there are not enough resources in the cluster to schedule
+`pod-e`.  So, the drain  then the drain will block.  The cluster may end up in this
+state:
+
+|    node-1 *drained*  |       node-2        |       node-3       | *no node*          |
+|:--------------------:|:-------------------:|:------------------:|:------------------:|
+|                      | pod-b *available*   | pod-c *available*  | pod-e *pending*    |
+|                      | pod-d *available*   | pod-y              |                    |
+
+At this point, the cluster administrator needs to
+add a node back to the cluster to proceed with the upgrade.
+
+You can see how Kubernetes determine the rate at which disruptions, like node upgrades,
```

Comment:
I must say, this section is very patient and detailed.  Good job.  I would not have thought to spell it out quite so clearly, even though I'm now certain the audience is going to profit from it big time.